chore(data): Add HealthDCAT-AP semantics

YpeZ opened the pull request:
In this branch we are working towards finalizing compatibility of our data model to the Health DCAT AP standard.
Columns of tables in our data model do not always map one to one to properties of classes in extended DCAT models. This PR therefore addresses the main holes in the road that must be fixed to achieve complete compatibility.


- [ ] check profiles column
- [ ] check mandatory vs non-mandatory and report inconsistencies in the manual of healthdcat-ap

YpeZ committed:
added properties "applicable legislation" and "access rights"

YpeZ committed:
added "provenance", "purpose"

YpeZ committed:
Merge branch 'master' into chore/data-catalogue-semantics

YpeZ committed:
renamed column 'default' to 'defaultValue'

BrendaHijmans commented on data/_models/shared/Resources.csv:
```diff
@@ -1,208 +1,212 @@
+Resources,,population of interest other,text,,,,,,,,,"If population of interest is 'Other', please specify which other population has been studied","DataCatalogueFlat,EMA,StudiesStaging",
+Resources,,countries,ontology_array,,,CatalogueOntologies,Countries,,,,dcterms:spatial,Countries where data/samples from this resource largely originate from,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,regions,ontology_array,,,CatalogueOntologies,Regions,,,,dcterms:spatial,Geographical regions where data/samples from this resource largely originate from,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,population age groups,ontology_array,,,CatalogueOntologies,Age groups,,,,"healthDCAT-AP:minimumTypicalAge,healthDCAT-AP:maximumTypicalAge",Which population age groups are captured in this resource? Select all that are relevant,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
```
split in min and max age

AJKellmann committed:
Updated the datamodel

to be checked, maybe nott correct

AJKellmann committed:
Undo the changes in dataset

Those adaptations should have happened to datasets not dataset.

AJKellmann committed:
Modifying models

Not the final version, but a base for discussion. Still working on those files.
Especially the datasets file still contains duplicate rows of the same colums  that differ slightly.

AJKellmann committed:
Update Datasets.csv

Datasets.csv > dcat:Distrbution

AJKellmann committed:
Update Datasets.csv

AJKellmann committed:
Update Datasets.csv

AJKellmann committed:
Update Resources.csv

Annotated Ressources.csv usind DcatAP's dcat:Dataset

BrendaHijmans commented on data/_models/shared/Datasets.csv:
```diff
@@ -1,14 +1,20 @@
+Datasets,,mapped from,refback,,,,Dataset mappings,,target dataset,,,Source datasets that have been mapped to this harmonised dataset,"DataCatalogueFlat,CohortsStaging,RWEStaging,StudiesStaging",,
+Datasets,,since version,,,,,,,,,,"When this dataset was introduced, if applicable","DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging",,
+Datasets,,until version,,,,,,,,,,"When this dataset was removed, if applicable","DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging",,
+Datasets,,accessUrl,hyperlink_array,,TRUE,,,,,,dcat:accessURL,"A URL of a resource that gives access to a distribution of the dataset. E.g. landing page, feed, SPARQL endpoint. Use for all cases except a simple download link, in which case downloadURL is preferred.",DCAT,,
```
change to 'access url'

BrendaHijmans commented on data/_models/shared/Datasets.csv:
```diff
@@ -1,14 +1,20 @@
+Datasets,,until version,,,,,,,,,,"When this dataset was removed, if applicable","DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging",,
+Datasets,,accessUrl,hyperlink_array,,TRUE,,,,,,dcat:accessURL,"A URL of a resource that gives access to a distribution of the dataset. E.g. landing page, feed, SPARQL endpoint. Use for all cases except a simple download link, in which case downloadURL is preferred.",DCAT,,
+Datasets,,applicable legislation,hyperlink_array,,TRUE,,,,,,dcatap:applicableLegislation,"Legislation applicable to the catalogue",DCAT,,http://data.europa.eu/eli/reg/2022/868/oj
+Datasets,,format,ontology,,TRUE,,,http://publications.europa.eu/resource/authority/file-type,,,dct:format,"Recommended practice is to use a controlled vocabulary where available. For example, for file formats one could use the list of Internet Media Types [[MIME](https://www.iana.org/assignments/media-types/media-types.xhtml)].  Examples of dimensions include size and duration.",DCAT,,
```
I cannot find this in healthdcat-ap

BrendaHijmans commented on data/_models/shared/Resources.csv:
```diff
@@ -1,208 +1,217 @@
+Resources,,name,text,3,TRUE,,,,,,dct:title,Name used in international projects,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,local name,,,,,,,,,rdfs:label,"If different from above, name in the national language","DataCatalogueFlat,CohortsStaging,RWEStaging",
+Resources,,acronym,,,,,,,,,dct:alternative,"Acronym, if applicable","DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,type,ontology_array,,TRUE,CatalogueOntologies,Resource types,,,,dct:type,"Type of resource, e.g. registry, cohort, biobank","DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
```
type will have a recommended controlled vocabulary in healthdcat-ap. For health datasets containing personal level information, the type of the dataset MUST take the value "personal data”.

BrendaHijmans replied:
healthdcatap:healthCategory

BrendaHijmans commented on data/_models/shared/Resources.csv:
```diff
@@ -1,208 +1,217 @@
+Resources,,modified,datetime,,,,,,,,dct:modified,This attribute is read only and written during migration of data from staging areas and used to produce valid dcat output. When the metadata for this resource was last modified,DataCatalogueFlat,
+Resources,,design and structure,heading,,,,,,,,,Structure of the resource,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,design,ontology,,,CatalogueOntologies,Cohort designs,,,,,"The study design of this cohort, i.e. cross-sectional or longitudinal","DataCatalogueFlat,CohortsStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
+Resources,,design description,text,,,,,,,,,"Short description of the study design of this resource, if applicable","DataCatalogueFlat,EMA,RWEStaging,CohortsStaging,UMCGCohortsStaging,UMCUCohorts,INTEGRATE",
```
this can be annotated: dcterms:provenance

AJKellmann committed:
Added most of the requested changes

BrendaHijmans commented:
this is outdated, most of healthdcat-ap mandatory items are now included in catalogue data model 7.0.0
